File: service/src/main/java/hu/bsstudio/robonaut/team/TeamService.java
package hu.bsstudio.robonaut.team;

import hu.bsstudio.robonaut.team.model.DetailedTeam;
import hu.bsstudio.robonaut.team.model.Team;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface TeamService {
    Mono<DetailedTeam> addTeam(final Team team);

    Mono<DetailedTeam> updateTeam(final Team team);

    Flux<DetailedTeam> findAllTeam();
}

File: server/service/src/main/kotlin/hu/bsstudio/robonaut/team/TeamService.kt
package hu.bsstudio.robonaut.team

import hu.bsstudio.robonaut.team.model.DetailedTeam
import hu.bsstudio.robonaut.team.model.Team
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface TeamService {
    fun addTeam(team: Team): Mono<DetailedTeam>

    fun updateTeam(team: Team): Mono<DetailedTeam>

    fun updateTeam(detailedTeam: DetailedTeam): Mono<DetailedTeam>

    fun findAllTeam(): Flux<DetailedTeam>
}
